src/OpenFeature.Contrib.Providers.Flagd/Resolver/InProcess/InProcessResolver.cs

```diff
@@ -0,0 +1,288 @@
+using System.Net.Http;
+using System.Threading.Tasks;
+using OpenFeature.Model;
+using OpenFeature.Flagd.Grpc.Sync;
+using System;
+using System.IO;
+using System.Linq;
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+using Grpc.Net.Client;
+using System.Net.Sockets;
+using System.Threading;
+using Grpc.Core;
+using Value = OpenFeature.Model.Value;
+
+namespace OpenFeature.Contrib.Providers.Flagd.RResolver.InProcess
+{
+    internal class InProcessResolver : Resolver.Resolver
+    {
+        static readonly int EventStreamRetryBaseBackoff = 1;
+        static readonly int MaxEventStreamRetryBackoff = 60;
+
+        static readonly CancellationTokenSource _cancellationTokenSource = new CancellationTokenSource();
+
+        private readonly FlagSyncService.FlagSyncServiceClient _client;
+        private readonly JsonEvaluator _evaluator;
+
+        private readonly Mutex _mtx;
+        private int _eventStreamRetryBackoff = EventStreamRetryBaseBackoff;
+
+        private readonly FlagdConfig _config;
+        private Thread _handleEvents;
+
+        internal InProcessResolver(FlagdConfig config)
+        {
+            _config = config;
+            _client = BuildClient(config, channel => new FlagSyncService.FlagSyncServiceClient(channel));
+            _mtx = new Mutex();
+            _evaluator = new JsonEvaluator(config.SourceSelector);
+        }
+
+        internal InProcessResolver(FlagSyncService.FlagSyncServiceClient client, FlagdConfig config)
```

Comment:
Minor - please check if this is used (seems not per my view). And if it is, let's use constructor chaining to avoid duplicate code :)

Reply:
good catch - it is used in the unit tests, but i adapted the code to use a constructor chain :)